Source/Urho3D/UI/UIElement.cpp

```diff
@@ -887,63 +887,63 @@ void UIElement::SetUseDerivedOpacity(bool enable)
 void UIElement::SetEnabled(bool enable)
 {
     enabled_ = enable;
-    enabledPrev_ = enable;
+    enabledSelf_ = enable;
 }
 
 void UIElement::SetDeepEnabled(bool enable)
 {
     enabled_ = enable;
 
-    for (Vector<SharedPtr<UIElement> >::ConstIterator i = children_.Begin(); i != children_.End(); ++i)
-        (*i)->SetDeepEnabled(enable);
+    for (SharedPtr<UIElement> child : children_)

```

Comment:
It should be either `const SharedPtr<UIElement>& child` or `UIElement*`, since copying of `SharedPtr` is relatively slow operation.

I know that performance impact is negligibly small (since this is user API of UI subsystem), but it doesn't mean we should do premature pessimization when alernatives exist.